packages/ast/src/lowering.candy

```diff
@@ -1370,20 +1432,333 @@ fun lowerInterpolatedStringPart(
+}
+
+fun lowerTupleExpression(
+  context: LoweringContext,
+  parentId: AstDeclarationId | AstBodyId | AstValueParameterId | AstExpressionId,
+  cst: CstNode<CstTupleExpression>,
+): Maybe<AstExpression> {
+  let id = context.getExpressionId(parentId)
+  mut let isValid = true
+
+  let expressions = MutableList.empty<AstExpression>()
+  mut let punctuationExpectedNext = false
+  for node in cst.child.expressions {
+    if node.child is PunctuationToken {
+      if !punctuationExpectedNext {
+        // TODO(JonasWanke)
+      } else {
+        punctuationExpectedNext = false
+      }
+    } else {
+      if punctuationExpectedNext {
+        // TODO(JonasWanke)
+      }
+      let type = lowerExpression(context, parentId, node.cast<CstExpression>())
+      if type is Some<AstExpression> { expressions.append(type.unwrap()) }
+      punctuationExpectedNext = true
+    }
+  }
+  if (expressions as Iterable<AstExpression>).length() < 2 {
+    isValid = false
+    context.addError(
+      ExpectedTwoOrMoreItemsInTupleCompilerError(
+        context.file,
+        cst.cast<CstTupleType | CstTupleExpression>(),
+      ),
+    )
+  }
+
+  if cst.child.closingParenthesis is None<CstNode<PunctuationToken>> {
+    let upcastedCst = cst.cast<CstDeclarationContent
+        | CstBlockBody
+        | CstValueParameters
+        | CstGroupType
+        | CstFunctionTypeParameterTypes
+        | CstTupleType
+        | CstTypeParameters
+        | CstTypeArguments
+        | CstGroupExpression
+        | CstTupleExpression
+        | CstInterpolatedStringPart
+        | CstValueArguments>()
+    context.addError(MissingClosingBracketCompilerError(context.file, upcastedCst))
+  }
+
+  if !isValid { return None<AstExpression>() }
+  context.addIdMapping(cst.id, id)
+  Some<AstExpression>(AstTupleExpression(id, expressions))
+}
+
+fun lowerBinaryExpression(
+  context: LoweringContext,
+  parentId: AstDeclarationId | AstBodyId | AstValueParameterId | AstExpressionId,
+  cst: CstNode<CstBinaryExpression>,
+): Maybe<AstExpression> {
+  let id = context.getExpressionId(parentId)
+
+  if cst.child.leftOperand is None<CstNode<CstExpression>>
+    || cst.child.rightOperand is None<CstNode<CstExpression>> {
+    context.addError(
+      ExpectedExpressionCompilerError(
+        context.file,
+        cst.cast<CstExpressionBody | CstDelegationBody | CstGroupExpression | CstBinaryExpression | CstValueArgument>(),
+      )
+    )
+  }
+
+  let leftOperand = cst.child.leftOperand
+      .flatMap<AstExpression>({ lowerExpression(context, id, it) })
+
+  let operatorId = AstIdentifierId(id, 0)
+  context.addIdMapping(cst.child.operator.id, operatorId)
+  let operator = AstIdentifier(operatorId, cst.child.operator.child.punctuation)
+
+  let rightOperand = cst.child.rightOperand
+      .flatMap<AstExpression>({ lowerExpression(context, id, it) })
+
+  context.addIdMapping(cst.id, id)
+  Some<AstExpression>(AstBinaryExpression(id, leftOperand, operator, rightOperand))
```

Comment:
How does this work? If I'm not mistaken, the operands are still wrapped in a `Maybe` and the `AstBinaryExpression` should require actual expressions, shouldn't it?